### PR TITLE
[ST] Refactor INFRA_NAMESPACE to use clusterOperator deploymentNamespace

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

_Select the type of your PR_

- Refactoring

### Description
Do not merge. Just for testing purposes - for now.

This PR is based around refactoring INFRA_NAMESPACE to avoid hard-coded value in every test. It's better to use clusterOperator and it's deploymentNamespace. This way, any namespace change can be easily achieved without modifying every instance of INFRA_NAMESPACE constant.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainIsolatedST.java [hunk 1]
```diff
@@ -274,16 +273,16 @@ void testProducerConsumerConnect(ExtensionContext extensionContext) {
...
         resourceManager.createResource(extensionContext, oauthExampleClients.producerStrimziOauthPlain());
-        ClientUtils.waitForClientSuccess(producerName, INFRA_NAMESPACE, MESSAGE_COUNT);
+        ClientUtils.waitForClientSuccess(producerName, clusterOperator.getDeploymentNamespace(), MESSAGE_COUNT);
 
         resourceManager.createResource(extensionContext, oauthExampleClients.consumerStrimziOauthPlain());
-        ClientUtils.waitForClientSuccess(consumerName, INFRA_NAMESPACE, MESSAGE_COUNT);
+        ClientUtils.waitForClientSuccess(consumerName, clusterOperator.getDeploymentNamespace(), MESSAGE_COUNT);
 
-        KafkaConnect connect = KafkaConnectTemplates.kafkaConnectWithFilePlugin(extensionContext, clusterName, INFRA_NAMESPACE, oauthClusterName, 1)
+        KafkaConnect connect = KafkaConnectTemplates.kafkaConnectWithFilePlugin(extensionContext, clusterName, Constants.INFRA_NAMESPACE, oauthClusterName, 1)
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainIsolatedST.java:283]: Why can't you use the `clusterOperator.getDeploymentNamespace()` here?

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainIsolatedST.java [hunk 2]
```diff
@@ -314,15 +313,15 @@ void testProducerConsumerConnect(ExtensionContext extensionContext) {
 
         resourceManager.createResource(extensionContext, connect);
 
-        final String kafkaConnectPodName = kubeClient(INFRA_NAMESPACE).listPods(INFRA_NAMESPACE, clusterName, Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND).get(0).getMetadata().getName();
+        final String kafkaConnectPodName = kubeClient(clusterOperator.getDeploymentNamespace()).listPods(clusterOperator.getDeploymentNamespace(), clusterName, Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND).get(0).getMetadata().getName();
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainIsolatedST.java:316]:
```suggestion
        final String kafkaConnectPodName = kubeClient().listPods(clusterOperator.getDeploymentNamespace(), clusterName, Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND).get(0).getMetadata().getName();
```

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeIsolatedST.java [hunk 3]
```diff
@@ -202,19 +200,19 @@ void testClientScopeKafkaSetIncorrectly(ExtensionContext extensionContext) throw
 
             ((KafkaListenerAuthenticationOAuth) scopeListeners.get(0).getAuth()).setClientScope(null);
             kafka.getSpec().getKafka().getListeners().set(0, scopeListeners.get(0));
-        }, INFRA_NAMESPACE);
+        }, clusterOperator.getDeploymentNamespace());
 
-        kafkaPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(INFRA_NAMESPACE, kafkaSelector, 3, kafkaPods);
+        RollingUpdateUtils.waitForComponentAndPodsReady(clusterOperator.getDeploymentNamespace(), kafkaSelector, 1);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthScopeIsolatedST.java:205]: Why did you change this one?

File: systemtest/src/test/java/io/strimzi/systemtest/specific/HelmChartIsolatedST.java [hunk 4]
```diff
@@ -23,16 +23,16 @@
...
+import static io.strimzi.systemtest.Constants.INFRA_NAMESPACE;
 
 @Tag(HELM)
 @Tag(REGRESSION)
 @IsolatedSuite
 class HelmChartIsolatedST extends AbstractST {
 
     private static final Logger LOGGER = LogManager.getLogger(HelmChartIsolatedST.class);
-    private HelmResource helmResource = new HelmResource(INFRA_NAMESPACE);
+    private HelmResource helmResource = new HelmResource(clusterOperator.getDeploymentNamespace());
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/specific/HelmChartIsolatedST.java:35]: Will this work when the `clusterOperator` is not initialized?

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainIsolatedST.java (hunk 2 above)
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainIsolatedST.java:316]: will do on every occurence

**Review by im-konge — APPROVED** [review]
LGTM, assuming it works.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicScalabilityIsolatedST.java [hunk 5]
```diff
@@ -38,56 +38,53 @@ public class TopicScalabilityIsolatedST extends AbstractST {
...
     }
 
     @IsolatedTest
     void testModifyBigAmountOfTopicPartitions(ExtensionContext extensionContext) {
         final int defaultPartitionCount = 2;
-        final String namespace = INFRA_NAMESPACE;
 
         // Create topics
-        KafkaTopicScalabilityUtils.createTopicsViaK8s(extensionContext, namespace, sharedClusterName, topicPrefix,
+        KafkaTopicScalabilityUtils.createTopicsViaK8s(extensionContext, clusterOperator.getDeploymentNamespace(), sharedClusterName, topicPrefix,
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicScalabilityIsolatedST.java:55]: Are you sure this is intended?

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicScalabilityIsolatedST.java:55]: Discussed offline, it's ok

**Review by Frawless — APPROVED** [review]
